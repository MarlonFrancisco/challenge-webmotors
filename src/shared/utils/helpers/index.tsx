import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styled";

type WithThemeProps = {};

export const WithTheme: React.FC<WithThemeProps> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
