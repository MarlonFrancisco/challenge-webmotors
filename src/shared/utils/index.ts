export type Options = {
  withoutPrefix: boolean;
  splitted: boolean;
};

export const formatCurrency = (
  value = 0,
  options?: Options
): string | string[] => {
  const formatted = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
  }).format(value);
  let apllyOptions: string | string[] = formatted;

  if (!options) {
    return formatted;
  }

  if (options.withoutPrefix) {
    apllyOptions = apllyOptions.replace("R$", "").trim();
  }

  if (options.splitted) {
    apllyOptions = apllyOptions.split(",");
  }

  return apllyOptions;
};
