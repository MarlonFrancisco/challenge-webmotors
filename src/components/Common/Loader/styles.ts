import styled from "styled-components";
import { Theme } from "../../../shared/@types";

export const Spin = styled.div`
  border: 3px solid #f3f3f3;
  border-top: 3px solid
    ${(props: { theme: Theme }) => props.theme.palette.secondary.main};
  border-radius: 50%;
  width: ${(props: { size: number }) => props.size}px !important;
  margin: auto;
  height: ${(props: { size: number }) => props.size}px;
  animation: spin 2s linear infinite;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;
