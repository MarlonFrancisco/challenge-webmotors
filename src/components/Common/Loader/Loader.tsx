import React from "react";
import { Spin } from "./styles";

export type LoaderProps = {
  size?: number;
};

const Loader: React.FC<LoaderProps> = ({ size = 30 }) => (
  <Spin data-testid="loader-test" size={size} />
);

export default Loader;
