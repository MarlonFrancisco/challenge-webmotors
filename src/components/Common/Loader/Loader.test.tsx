import React from "react";
import { render } from "@testing-library/react";
import Loader, { LoaderProps } from "./Loader";
import { WithTheme } from "../../../shared/utils/helpers";

const Wrapper = (props: LoaderProps) => (
  <WithTheme>
    <Loader {...props} />
  </WithTheme>
);

describe("Loader", () => {
  let props: LoaderProps;

  beforeEach(() => {
    props = {};
  });

  it("should render without an error", () => {
    const { container } = render(<Wrapper {...props} />);
    expect(container).toMatchSnapshot();
  });
});
