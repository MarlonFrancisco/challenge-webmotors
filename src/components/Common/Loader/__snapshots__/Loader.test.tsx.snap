// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render without an error 1`] = `
<div>
  <div
    class="sc-bdnylx bmMCJM"
    data-testid="loader-test"
    size="30"
  />
</div>
`;
